XiEditor/EditView.swift

```diff
@@ -42,19 +42,13 @@ struct TextDrawingMetrics {
     
     init(font: NSFont) {
         self.font = font
-        ascent = CTFontGetAscent(font)
-        descent = CTFontGetDescent(font)
-        leading = CTFontGetLeading(font)
+        ascent = font.ascender
+        descent = abs(font.descender)
```

Comment:
Any reason for choosing `abs` here? I suspect there are situations where this could cause `linespace` to be incorrect.